scripts/jenkins/cloud/ansible/roles/list_packages/tasks/list-packages.yml

```
@@ -31,7 +31,14 @@
 
 - name: Collect list of installed packages repo of origin from zypper
   shell: |
-    zypper --xmlout se -s -i -t package
+    if [ $(grep -c SUSE /etc/issue) -eq 1 ];then
```

I would prefer to not count the occurrences of "SUSE" but to simply check whether that string is present or not. Who knows, maybe at some point `/etc/issue` will contain 2 "SUSE"s :smile:

```suggestion
    if grep --quiet SUSE /etc/issue; then
```

---

you also have the option to use ansible facts to check in which OS the task is running on.

---

I think I prefer using the ansible facts. Check out these tasks from Ardana's osconfig-ansible repository:

```- name: osconfig | udev config | Set the unit file for RedHat
  become: yes
  template: src=udev-unit.j2 dest=/lib/systemd/system/ardana-udev.service
  when: ansible_os_family == 'RedHat'

- name: osconfig | udev config | Set the unit file for SUSE
  become: yes
  template: src=udev-unit.j2 dest=/usr/lib/systemd/system/ardana-udev.service
  when: ansible_os_family == 'Suse'```

---

@flaviodsr  @stefannica I tried to use facts but sometimes it didn t show correctly(it showed local os rather then target one). I can try that again.

---

I agree with using ansible facts ...
(But if you want to check some file on a system, why not use /etc/os-release?)

---

I think I found the way with gather_facts